src/Kathara/manager/kubernetes/KubernetesMachine.py

```diff
@@ -458,11 +458,17 @@ def _build_definition(self, machine: Machine, config_map: client.V1ConfigMap) ->
                 )
             ))
 
+        # Add image_pull_secrets if using a private registry
+        image_pull_secrets_arg = {}
+        if Setting.get_instance().private_registry_dockerconfigjson:
+            image_pull_secrets_arg = {"image_pull_secrets": [client.V1LocalObjectReference(name="private-registry")]}
```

Comment:
If you move the logic to the `KubernetesSecret` class, this if statement would be the place where you create the Secret.

Reply:
This part of the code is required to enable the pod to use the private registries defined in the secret. It is a field in the pod definition which is separate from the secret.

I think the best solution is to create the secret (if needed) right after the namespace creation (inside the `create` function in `KubernetesNamespace`).

This is also because the secret is only one, but on each pod we need to add the `imagePullSecrets` property to allow access to the private registries ([documentation](https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod)).